packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -146,6 +146,170 @@ class _OpenUpwardsPageTransition extends StatelessWidget {
   }
 }
 
+// Zooms and fades a new page in, zooming out the previous page. This transition
+// is designed to match the Android 10 activity transition.
+class _ZoomPageTransition extends StatefulWidget {
+  const _ZoomPageTransition({
+    Key key,
+    this.animation,
+    this.secondaryAnimation,
+    this.child,
+  }) : super(key: key);
+
+  // The scrim obscures the old page by becoming increasingly opaque.
+  static final Tween<double> _scrimOpacityTween = Tween<double>(
+    begin: 0.0,
+    end: 0.60,
+  );
+
+  // A curve sequence similar to the 'fastOutExtraSlowIn' curve used in the
+  // native transition.
+  static final List<TweenSequenceItem<double>> fastOutExtraSlowInTweenSequenceItems = <TweenSequenceItem<double>>[
+    TweenSequenceItem<double>(
+      tween: Tween<double>(begin: 0.0, end: 0.4)
+        .chain(CurveTween(curve: const Cubic(0.05, 0.0, 0.133333, 0.06))),
+      weight: 0.166666,
+    ),
+    TweenSequenceItem<double>(
+      tween: Tween<double>(begin: 0.4, end: 1.0)
+        .chain(CurveTween(curve: const Cubic(0.208333, 0.82, 0.25, 1.0))),
+      weight: 1.0 - 0.166666,
+    ),
+  ];
+  static final TweenSequence<double> _scaleCurveSequence = TweenSequence<double>(fastOutExtraSlowInTweenSequenceItems);
+  static final FlippedTweenSequence _flippedScaleCurveSequence = FlippedTweenSequence(fastOutExtraSlowInTweenSequenceItems);
+
+  final Animation<double> animation;
+  final Animation<double> secondaryAnimation;
+  final Widget child;
+
+  @override
+  __ZoomPageTransitionState createState() => __ZoomPageTransitionState();
+}
+
+class __ZoomPageTransitionState extends State<_ZoomPageTransition> {
+  AnimationStatus _currentAnimationStatus;
+  AnimationStatus _lastAnimationStatus;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.animation.addStatusListener((AnimationStatus animationStatus) {
+      _lastAnimationStatus = _currentAnimationStatus;
+      _currentAnimationStatus = animationStatus;
+    });
+  }
+
+  bool get _transitionWasInterrupted {
```

Comment:
This would be a good place to explain why we're checking if the transition was interrupted.